src/main/java/frc/LoggyThings/ILoggyMotor.java

```diff
@@ -48,112 +64,133 @@ public class DataLogEntryWithHistory extends DataLogEntry {
             }
         }
 
+        /**
+         * Logs a boolean and provides context if the value has changed
+         * 
+         * @param newValue the boolean to be logged
+         * @param now      a long value representing the time
+         */
         void logBooleanIfChanged(boolean newValue, long now) {
-            if ((((Boolean) mLastValue).booleanValue() != newValue)||(now>(lastLogTime+500000))) {
-                m_log.appendDouble(m_entry, newValue?1.0:0, now);
+            if ((((Boolean) mLastValue).booleanValue() != newValue) || (now > (lastLogTime + 500000))) {
```

Comment:
Could just make this a function instead of having the same `|| (now > (lastLogTime + 500000)` everywhere

Reply:
does it make a difference in terms of processing speed?

Reply:
Probably no difference performance wise, but it makes the code more readable. Also, your `hasHalfSecondPassed` function not the same functionality as `(now > (lastLogTime + 500000)` as the input `now` could be different from `WPIUtilJNI.now()`